src/log_surgeon/finite_automata/RegexAST.hpp

```diff
@@ -667,12 +813,55 @@ void RegexASTMultiplication<NFAStateType>::add_to_nfa(
     nfa->set_root(saved_root);
 }
 
+template <typename NFAStateType>
+auto RegexASTMultiplication<NFAStateType>::assign_negative_and_collect_positive_descendent_tags(
+) -> std::set<uint32_t> {
+    return m_operand->assign_negative_and_collect_positive_descendent_tags();
+}
+
+template <typename NFAStateType>
+auto RegexASTMultiplication<NFAStateType>::serialize(bool const with_tags) -> std::string {
+    std::string serialized_string
+            = m_operand->serialize(with_tags) + "{" + std::to_string(m_min) + ",";
+    if (is_infinite()) {
+        serialized_string += "inf";
+    } else {
+        serialized_string += std::to_string(m_max);
+    }
+    serialized_string += "}";
+    if (with_tags) {
+        serialized_string += this->serialize_negative_tags();
+    }
+    return serialized_string;
+}
+
 template <typename NFAStateType>
 void RegexASTCapture<NFAStateType>::add_to_nfa(RegexNFA<NFAStateType>* nfa, NFAStateType* end_state)
         const {
     m_group_regex_ast->add_to_nfa(nfa, end_state);
 }
 
+template <typename NFAStateType>
+auto RegexASTCapture<NFAStateType>::assign_negative_and_collect_positive_descendent_tags(
+) -> std::set<uint32_t> {
+    auto tags = m_group_regex_ast->assign_negative_and_collect_positive_descendent_tags();
+    tags.insert(m_tag);
+    return tags;
+}
+
+template <typename NFAStateType>
+auto RegexASTCapture<NFAStateType>::serialize(bool const with_tags) -> std::string {
+    std::string serialized_string = "(";
+    if (false == with_tags) {
+        serialized_string += "?<" + m_group_name + ">";
+    }
+    serialized_string += m_group_regex_ast->serialize(with_tags) + ")";
+    if (with_tags) {
+        serialized_string += "<" + std::to_string(m_tag) + ">" + this->serialize_negative_tags();
+    }
+    return serialized_string;
```

Comment:
```suggestion
    std::string serialized_string;
    if (with_tags) {
        serialized_string = "(" + m_group_regex_ast->serialize(with_tags) + ")<"
                            + std::to_string(m_tag) + ">" + this->serialize_negative_tags();
    } else {
        serialized_string
                = "(?<" + m_group_name + ">" + m_group_regex_ast->serialize(with_tags) + ")";
    }
    return serialized_string;
```
I think having `if (false == with_tags)` and `if (with_tags)` like this is hard to read, how about let's try to be verbose? Also I think a lot of these format stuff can be optimized by using `fmt` library

Reply:
Removed the `with_tags` argument. Will look into `fmt` library.

Reply:
Good call on the fmt library. Changed all serialize functions to use it.

Reply:
Pushed.